tests/test_graph.py

```diff
@@ -158,6 +159,93 @@ def test_disconnected_graph(self) -> None:
+        assert path
+        self.assertEqual(list(path), [v1])
+
+    def test_dijkstra_search_with_triangle_graph1(self) -> None:
+        test_vertexes: Dict[int, Vertex[int]] = TestGraph._generate_graph([1, 2, 4], [(1, 2), (2, 3), (1, 3)])
+
+        g1 = Graph[int](list(test_vertexes.values()))
+        path = g1.find_weighted_shortest_path(test_vertexes[1], test_vertexes[3])
+
+        assert path
+        self.assertEqual(list(path), [test_vertexes[3], test_vertexes[2], test_vertexes[1]])
+
+    def test_dijkstra_search_with_triangle_graph2(self) -> None:
+        test_vertexes: Dict[int, Vertex[int]] = TestGraph._generate_graph([1, 2, 2], [(1, 2), (2, 3), (1, 3)])
+
+        g1 = Graph[int](list(test_vertexes.values()))
+        path = g1.find_weighted_shortest_path(test_vertexes[1], test_vertexes[3])
+
+        assert path
+        self.assertEqual(list(path), [test_vertexes[3], test_vertexes[1]])
+
+    def test_dijkstra_search_with_triangle_graph3(self) -> None:
+        test_vertexes: Dict[int, Vertex[int]] = TestGraph._generate_graph([1, 2, 3], [(1, 2), (2, 3), (1, 3)])
+
+        g1 = Graph[int](list(test_vertexes.values()))
+        path = g1.find_weighted_shortest_path(test_vertexes[1], test_vertexes[3])
+
+        assert path
+        self.assertEqual(list(path), [test_vertexes[3], test_vertexes[1]])
+
+    def test_dijkstra_search_with_cycle(self) -> None:
+        test_vertexes: Dict[int, Vertex[int]] = TestGraph._generate_graph([1, 1, 1, 1, 1], [(1, 2), (2, 3), (3, 4), (4, 5), (5, 2)])
+
+        g1 = Graph[int](list(test_vertexes.values()))
+        path = g1.find_weighted_shortest_path(test_vertexes[1], test_vertexes[5])
+
+        assert path
+        self.assertEqual(list(path), [test_vertexes[5], test_vertexes[4], test_vertexes[3], test_vertexes[2], test_vertexes[1]])
+
+    def test_dijkstra_search_with_islands(self) -> None:
+        test_vertexes: Dict[int, Vertex[int]] = TestGraph._generate_graph([1, 1, 1, 2, 2, 2], [(1, 2), (2, 3), (3, 1), (4, 5), (5, 6), (6, 4)])
+
+        g1 = Graph[int](list(test_vertexes.values()))
+        self.assertFalse(g1.dijkstra_search(test_vertexes[1], test_vertexes[4]))
+
+        path = g1.find_weighted_shortest_path(test_vertexes[1], test_vertexes[4])
+
+        self.assertIsNone(path)
+
+    def test_dijkstra_search_two_zones(self) -> None:
+        test_vertexes: Dict[int, Vertex[int]] = TestGraph._generate_graph(
+            [1, 2, 3, 1, 1, 1, 1, 2, 3, 1, 1, 1], [(1, 2), (1, 3), (1, 4), (2, 5), (3, 5), (4, 5), (5, 6), (5, 7), (5, 8), (6, 9), (7, 9), (8, 9)]
+        )
+
+        g1 = Graph[int](list(test_vertexes.values()))
+        path = g1.find_weighted_shortest_path(test_vertexes[1], test_vertexes[9])
+
+        assert path
+        self.assertEqual(list(path), [test_vertexes[9], test_vertexes[6], test_vertexes[5], test_vertexes[2], test_vertexes[1]])
+
+    @classmethod
+    def _generate_graph(cls, weights: List[int], edges: List[Tuple[int, int]]) -> Dict[int, Vertex[int]]:
```

Comment:
I suggest returning directly a `Graph` object, without using an intermediate `Dict`

Reply:
How would you find the vertexes for the searches?

Reply:
This would still mean that you need sequentially numbered vertexes starting with 0. The point of the dictionary was that you can number the vertexes however you like. 

You could, in theory, number a certain zone's vertexes 901, 902, 903 for instance to make it easier to visualize. Maybe I'm just overthinking it. 

It would also require another value check in `_generate_graph()`

Reply:
If non-sequential vertex names were needed, one could easily build an ad-hoc dictionary instead of a list from the `vertexes` iterator: `vertexes = [v.name:v for v in graph.vertexes]`.

I thought some more about it. How about this:
* have `_generate_graph()` return a `Graph`
* create and return the `Graph` with `return Graph(sorted([v for k, v in vertexes]))`
* outside `_generate_graph()` access vertexes with `vertexes = list(graph.vertexes)` or `vertexes = [v.name:v for v in graph.vertexes]` (for the non-sequential case).

Does that work?

Reply:
One more suggestion: can we have the `_generate_graph()` receive one list with a ternary tuple (start, end, weight)? That reduces risk of having mismatched lists and makes it easier to "read" the graph without having to match positions. Sorry if I'm insisting on the `Graph` creation utility but I'd like to use it to add many more tests and I'd like it to be as intuitive and readable as possible.